internal/auth/certs.go

```diff
@@ -60,14 +59,31 @@ func CreateCertificate(ctx context.Context, cfg *rest.Config, name, namespace, s
 		origin = "local"
 	}
 
-	err := createCertificate(ctx, cfg, name, namespace, systemDomain, issuer)
+	obj, err := createCertificate(name, namespace, systemDomain, issuer)
 	if err != nil {
 		return errors.Wrap(err, fmt.Sprintf("creation of %s ssl certificate failed", origin))
 	}
+
+	client, err := cluster.ClientCertificate()
+	if err != nil {
+		return err
+	}
+
+	if owner != nil {
+		obj.SetOwnerReferences([]metav1.OwnerReference{*owner})
```

Comment:
Would it be possible to pass the owner to the `createCertificate` function and create the resource with the desired owner in one step?

Reply:
I wanted to separate functions that use a client to create objects from the funcs that construct the object being created. That way you keep construction and creation (?) separate. You don't need to pass too many arguments around that way and you keep the error handling, client, context (maybe logging) in one top level function.

Ah, I saw your comment on Slack. `SetOwnerReferences` doesn't reach out to the API server.

Reply:
Oh true, my mistake. All good.